src/Engine/Core/Effects/Effect.cs

```diff
@@ -55,17 +55,14 @@ public void SetFxParam<T>(string name, T value)
         /// </summary>
         /// <param name="hash">Hash of the uniform variable. Retrieved by name.GetHashCode().</param>
         /// <param name="value">Value of the uniform variable</param>
-        public void SetFxParam<T>(int hash, T value)
+        public void SetFxParam(int hash, object value)
```

Comment:
I'm not sure why we can't use generics anymore.

See:
https://stackoverflow.com/a/4424049

> Using object's results in cast operations and boxing/unboxing of value-types. Because of these reasons generics are faster